bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/AbstractAreaHandler.java

```diff
@@ -0,0 +1,205 @@
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.omnilink.internal.handler;
+
+import static org.openhab.binding.omnilink.internal.OmnilinkBindingConstants.*;
+
+import java.math.BigInteger;
+import java.text.MessageFormat;
+import java.util.EnumSet;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.omnilink.internal.AreaAlarm;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.digitaldan.jomnilinkII.Message;
+import com.digitaldan.jomnilinkII.MessageTypes.ObjectStatus;
+import com.digitaldan.jomnilinkII.MessageTypes.SecurityCodeValidation;
+import com.digitaldan.jomnilinkII.MessageTypes.statuses.AreaStatus;
+import com.digitaldan.jomnilinkII.MessageTypes.systemevents.AllOnOffEvent;
+import com.digitaldan.jomnilinkII.OmniInvalidResponseException;
+import com.digitaldan.jomnilinkII.OmniUnknownMessageTypeException;
+
+/**
+ * The {@link AbstractAreaHandler} defines some methods that can be used across
+ * the many different areas defined in an OmniLink Controller.
+ *
+ * @author Craig Hamilton - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractAreaHandler extends AbstractOmnilinkStatusHandler<AreaStatus> {
+    private final Logger logger = LoggerFactory.getLogger(AbstractAreaHandler.class);
+
+    public AbstractAreaHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand: {}, command: {}", channelUID, command);
+
+        switch (channelUID.getId()) {
+            case CHANNEL_AREA_ACTIVATE_KEYPAD_EMERGENCY:
+                handleKeypadEmergency(channelUID, command);
+                break;
+            default:
+                handleSecurityMode(channelUID, command);
+                break;
+        }
+    }
+
+    private void handleSecurityMode(ChannelUID channelUID, Command command) {
+        int mode = getMode(channelUID);
+        int areaNumber = getThingNumber();
+
+        if (!(command instanceof StringType)) {
+            throw new IllegalArgumentException(
+                    MessageFormat.format("Command {} is not valid for channel {}, only StringType are accepted",
+                            command, channelUID.getId()));
```

Comment:
As receiving invalid types should be treated as a normal use case, it seems unreasonable to throw an unchecked exception. You could simply log it. Same for below.